loky/process_executor.py

```diff
@@ -405,7 +409,7 @@ def shutdown_all_workers():
         with processes_management_lock:
             nb_children_alive = sum(p.is_alive() for p in processes.values())
         for i in range(0, nb_children_alive):
-            call_queue.put_nowait(None)
+            call_queue.put(None)
```

Comment:
Also, I am not sure this is the cause of the failure. If a worker dies with `call_queue.rlock`, this should be detected thru a different process and no sentinel should be sent as we kill all workers before reaching this point. The only possibility of locking here is if a worker dies when it receives the sentinel. This is extremely unlikely and I am not sure we want to be robust to this.  

The random failures were caused by the new behavior of `_thread_management_worker` that cause the `gc` on the executor to be a bit delayed. I added some patience for the gc to avoid these failures.

However, this `Queue` size issue is bad for the pre-dispatching and might cause performance drop. So it should still be adressed.